frappe/public/js/frappe/views/reports/query_report.js

```diff
@@ -217,18 +235,39 @@ frappe.views.QueryReport = class QueryReport extends frappe.views.BaseList {
 			callback: resolve
 		})).then(r => {
 			const data = r.message;
-
-			this.toggle_message(false);
-
-			if (data.result && data.result.length) {
-				this.render_chart(data);
-				this.render_report(data);
-			} else {
-				this.toggle_nothing_to_show(true);
+            if (data.background_report){
+                this.toggle_message(true, 'This report is background. You can run this by hitting Run in background');
+			}else{
+                this.toggle_message(false);
+                if (data.result && data.result.length) {
+                    this.render_chart(data);
+                    this.render_report(data);
+                } else {
+                    this.toggle_nothing_to_show(true);
+                }
 			}
 		});
 	}
-
+	
+	render_background_report() {
+		this.toggle_message(true);
+		const filters = this.get_filter_values(true);
+		return new Promise(resolve => frappe.call({
+			method: 'frappe.desk.query_report.background_enqueue_run',
+			type: 'GET',
+			args: {
+				report_name: this.report_name,
+				filters: filters
+			},
+			callback: resolve
+		})).then(r => {
+				const data = r.message;
+				this.toggle_nothing_to_show(true);
+				frappe
+				.msgprint("Background job initiated successfully. Track and access results  <a class='text-info' target='_blank' href="+data.redirect_url+">here</a>",
```

Comment:
Here, we can show a modal with the last "Prepared Report" and timestamp, so user can download it without opening a new doctype.